bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -215,7 +216,7 @@ private void addCallbackToListener(String sitemapName, String pageId, SitemapSub
     }
 
     private EList<Widget> collectWidgets(String sitemapName, String pageId) {
-        EList<Widget> widgets = null;
+        EList<Widget> widgets = new BasicEList<Widget>();
```

Comment:
fully makes sense!

Could you please also adapt (i.e. simplify) lines 206-215, not that we know it's never gonna be `null`?